src/server_manager/install_scripts/install_server.sh

```diff
@@ -343,9 +343,10 @@ function add_api_url_to_config() {
 }
 
 function check_firewall() {
-  local readonly GET_ACCESS_KEYS=$(curl --insecure -s ${LOCAL_API_URL}/access-keys)
-  local readonly GET_ACCESS_KEY_PORT="$DOCKER_CMD exec shadowbox node -e 'console.log($GET_ACCESS_KEYS[\"accessKeys\"][0][\"port\"])'"
-  local -r ACCESS_KEY_PORT=$($GET_ACCESS_KEY_PORT)
+  # Write the access keys to a temporary file due to the near-impossibility
```

Comment:
When you have shell escaping issues, you may resort to pipes and stdio/out.

In this case, you can pipe the curl output to the docker exec command, and tweak the node code to read from stdin. You can do that with `fs.readFileSync(0)`:
```
echo '{"accessKeys": [{"port": 100}]}' | node -r fs -p 'JSON.parse(fs.readFileSync(0)).accessKeys[0].port;'
```

That's safe than leaving an access key file behind on /tmp, which may be readable by others if you don't chmod. And it avoids a possible collision.

Reply:
Nice! I didn't realise it was so easy to read from stdin in Node :-)